### PR TITLE
Add instructions for using Trino with Polaris

### PR DESCRIPTION
**Pull request by colebow**
# Description

This adds documentation to the quickstart.md guide on using Polaris with the Trino query engine. It also references Trino as an option in README.md.

## Type of change

- [x] Documentation

# How Has This Been Tested?

This instructions have been tested locally to get Trino running with Polaris.

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [x] I have signed and submitted the [ICLA](https://github.com/polaris-catalog/polaris/blob/main/ICLA.md) and if needed, the [CCLA](https://github.com/polaris-catalog/polaris/blob/main/CCLA.md). See [Contributing](https://github.com/polaris-catalog/polaris/blob/main/CONTRIBUTING.md) for details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,9 +178,9 @@ $ curl -i -X PUT -H "Authorization: Bearer $PRINCIPAL_TOKEN" -H 'Accept: applica
   -d '{"name": "polaris", "id": 100, "type": "INTERNAL", "readOnly": false}'
 ```
 
-This creates a catalog called `polaris`. From here, you can use Spark to create namespaces, tables, etc.
+This creates a catalog called `polaris`. From here, you can use Spark or Trino to create namespaces, tables, etc.
```
Comment (flyrain) [inline README.md:181]: How about `any Iceberg REST compatible clients(e.g., Spark, Trino, PyIceberg)`?

Comment (colebow) [reply README.md]: Amended!

File: docs/quickstart.md [hunk 2]
```diff
@@ -321,3 +383,43 @@ spark.sql("SELECT * FROM quickstart_table").show(false)
...
 org.apache.iceberg.exceptions.ForbiddenException: Forbidden: Principal 'quickstart_user' with activated PrincipalRoles '[]' and activated ids '[6, 7]' is not authorized for op LOAD_TABLE_WITH_READ_DELEGATION
 ```
+
+### Connecting with Trino
+
+To use a Polaris-managed catalog in [Trino](https://trino.io/), you can
+configure Trino to use the Iceberg REST API.
+
+Generate and export a `PRINCIPAL_TOKEN` per the [README](/README.md#creating-a-catalog-manually),
+but create a Polaris catalog with the following specifications:
```
Comment (flyrain) [inline docs/quickstart.md:393]: but -> and

File: docs/quickstart.md [hunk 3]
```diff
@@ -321,3 +383,43 @@ spark.sql("SELECT * FROM quickstart_table").show(false)
...
+http://localhost:8181/api/management/v1/catalogs \
+-d '{"name": "polaris", "id": 100, "type": "INTERNAL", "readOnly": false, "storageConfigInfo": {"storageType": "FILE"}, "properties": {"default-base-location": "file:///tmp/path_to_catalog"}}'
+```
+
+You can then show catalogs with:
+
+```
+curl -i -X GET -H "Authorization: Bearer $PRINCIPAL_TOKEN" -H 'Accept: application/json' -H 'Content-Type: application/json' \
+http://localhost:8181/api/management/v1/catalogs
+```
```
Comment (flyrain) [inline docs/quickstart.md:406]: We don't have to go through the catalog creation process here, since it is covered above.

**Review by flyrain — COMMENTED** [review]
LGTM overall. Left minor comments. Thanks a lot to add the Trino part.

File: docs/quickstart.md [hunk 4]
```diff
@@ -321,3 +383,43 @@ spark.sql("SELECT * FROM quickstart_table").show(false)
...
+and you can follow [the Trino docs](https://trino.io/docs/current/installation.html)
+to install it.
+
+Once Trino is installed, create a catalog properties file, `polaris.properties`,
+in the `etc/catalog/` directory of your Trino installation. This is the file
+where you can configure Trino's Iceberg connector. Edit it to:
+
+```
+connector.name=iceberg
+iceberg.catalog.type=rest
```
Comment (ebyhr) [inline docs/quickstart.md:418]: Could you add the following config property? 
```
iceberg.rest-catalog.uri=http://localhost:8181/api/catalog
```

File: docs/quickstart.md [hunk 5]
```diff
@@ -321,3 +383,43 @@ spark.sql("SELECT * FROM quickstart_table").show(false)
...
+iceberg.rest-catalog.security=OAUTH2
+iceberg.rest-catalog.oauth2.token={the value of your PRINCIPAL_TOKEN}
+iceberg.rest-catalog.warehouse={the catalog name created in the above POST request, e.g. 'polaris'}
+```
+
+Start (or restart) Trino, and `SHOW CATALOGS` should show the Polaris catalog.
+You can then run `USE polaris` to access, query, or write to Polaris.
```
Comment (ebyhr) [inline docs/quickstart.md:425]: `USE polaris` won't work. The syntax is:
```
USE catalog.schema
USE schema
```
https://trino.io/docs/current/sql/use.html

File: README.md [hunk 6]
```diff
@@ -178,9 +178,10 @@ $ curl -i -X PUT -H "Authorization: Bearer $PRINCIPAL_TOKEN" -H 'Accept: applica
   -d '{"name": "polaris", "id": 100, "type": "INTERNAL", "readOnly": false}'
 ```
 
-This creates a catalog called `polaris`. From here, you can use Spark to create namespaces, tables, etc.
+This creates a catalog called `polaris`. From here, you can use any Iceberg REST
```
Comment (eric-maynard) [inline README.md:181]: Maybe we can leave out these style-only changes for now?

File: docs/quickstart.md [hunk 7]
```diff
@@ -321,3 +383,43 @@ spark.sql("SELECT * FROM quickstart_table").show(false)
...
+### Connecting with Trino
+
+To use a Polaris-managed catalog in [Trino](https://trino.io/), you can
+configure Trino to use the Iceberg REST API.
+
+Generate and export a `PRINCIPAL_TOKEN` per the [README](/README.md#creating-a-catalog-manually),
+but create a Polaris catalog with the following specifications:
+
+```
+curl -i -X POST -H "Authorization: Bearer $PRINCIPAL_TOKEN" -H 'Accept: application/json' -H 'Content-Type: application/json' \
```
Comment (eric-maynard) [inline docs/quickstart.md:396]: The quickstart uses the CLI, not the REST API directly

File: docs/quickstart.md (hunk 5 above)
Comment (eric-maynard) [inline docs/quickstart.md:425]: When I tried this, I had trouble creating a table. Let's make sure the quickstart contains a working end-to-end example

File: README.md (hunk 6 above)
Comment (colebow) [reply README.md:181]: They're in a different commit, so I can remove it if necessary. I figured it was generally an improvement though.

File: docs/quickstart.md [hunk 8]
```diff
@@ -321,3 +383,32 @@ spark.sql("SELECT * FROM quickstart_table").show(false)
...
+connector.name=iceberg
+iceberg.catalog.type=rest
+iceberg.rest-catalog.security=OAUTH2
+iceberg.rest-catalog.oauth2.token={the value of your PRINCIPAL_TOKEN}
+iceberg.rest-catalog.warehouse={your catalog name}
+iceberg.rest-catalog.uri=http://localhost:8181/api/catalog
+```
+
+Start (or restart) Trino, and `SHOW CATALOGS` should show the Polaris catalog.
+You can then run `USE catalogname.schemaname` to access, query, or write to
```
Comment (eric-maynard) [inline docs/quickstart.md:413]: Yeah, I'd love if this was end-to-end and involved creating a table and interacting with it. I noticed in my testing that Trino requires some additional permissions that are not granted above.

**Review by flyrain — CHANGES_REQUESTED** [review]
Thanks @colebow for working on it. Can you rebase it?

**Comment by cgpoh** [comment]
@colebow , for my trino configuration working with ADLS, I need to have the following properties:
```yaml
test_bronze: |-
        connector.name=iceberg
        iceberg.catalog.type=rest
        iceberg.rest-catalog.security=OAUTH2
        iceberg.rest-catalog.oauth2.credential=key:secret
        iceberg.rest-catalog.oauth2.scope=PRINCIPAL_ROLE:ALL
        iceberg.rest-catalog.warehouse=test_bronze
        iceberg.rest-catalog.vended-credentials-enabled=true
        iceberg.rest-catalog.uri=http://polaris-catalog.test.svc:8181/api/catalog
        fs.native-azure.enabled=true
        azure.auth-type=OAUTH
        azure.oauth.tenant-id=tenant-id
        azure.oauth.endpoint=https://login.microsoftonline.com/tenant-id/oauth2/token
        azure.oauth.client-id=client-id
        azure.oauth.secret=client-secret
```
Do we need a separate PR to address this ADLS Trino integration?

File: README.md (hunk 6 above)
Comment (aihuaxu) [reply README.md:181]: +1 to separate style-only changes to make review easier.

**Comment by mayankvadariya** [comment]
Do we still need this PR? given its already documented at https://github.com/apache/polaris/blob/main/getting-started/trino/README.md

**Pull request closed without merging**